src/System.IO.FileSystem/src/System/IO/File.cs

```diff
@@ -386,6 +403,76 @@ private static String InternalReadAllText(String path, Encoding encoding)
                 return sr.ReadToEnd();
         }
 
+
+        // If we use the path-taking constructors we will not have FileOptions.Asynchronous set and
+        // we will have asynchronous file access faked by the thread pool. We want the real thing.
+        private static StreamReader AsyncStreamReader(string path, Encoding encoding)
+        {
+            FileStream stream = new FileStream(
```

Comment:
@JonHanna, seems like it's here: [FileStream.cs#L190-L196](https://github.com/dotnet/corefx/blob/7454bb7/src/System.IO.FileSystem/src/System/IO/FileStream.cs#L190-L196) (GitHub search FTW :-))